modules/stochastic_tools/src/samplers/CartesianProductSampler.C

```diff
@@ -0,0 +1,61 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CartesianProductSampler.h"
+#include "Distribution.h"
+
+registerMooseObject("StochasticToolsApp", CartesianProductSampler);
+
+defineLegacyParams(CartesianProductSampler);
+
+InputParameters
+CartesianProductSampler::validParams()
+{
+  InputParameters params = Sampler::validParams();
+  params.addClassDescription("Provides complete cartesian product for the supplied variables.");
+  params.addRequiredParam<std::vector<Real>>(
+      "items", "A list of items, each item should include a min, step size, and number of steps.");
+  params.set<ExecFlagEnum>("execute_on") = EXEC_INITIAL;
+  return params;
+}
+
+CartesianProductSampler::CartesianProductSampler(const InputParameters & parameters)
+  : Sampler(parameters)
+{
+  const std::vector<Real> & items = getParam<std::vector<Real>>("items");
+  if (items.size() % 3 != 0)
+    paramError("items",
+               "The number of numeric items must be divisible by 3; min, max, divisions for each "
+               "item are required.");
+
+  std::vector<std::vector<Real>> grid_items;
+  for (std::size_t i = 0; i < items.size(); i += 3)
+  {
+    if (items[i + 2] != std::floor(items[i + 2]))
+      paramError("items", "The third entry for each item must be an integer.");
```

Comment:
Explain why it needs to be an integer ('because it defines the number of steps in the interval')